src/test/java/com/fasterxml/jackson/databind/cfg/CacheProviderTest.java

```diff
@@ -160,21 +182,84 @@ public void testDefaultCacheProviderSharesCache() throws Exception
         ObjectMapper mapper2 = JsonMapper.builder()
                 .cacheProvider(cacheProvider)
                 .build();
-        
+
         // Act 
         // 3. Add two different types to each mapper cache
         mapper1.readValue("{\"point\":24}", RandomBean.class);
         mapper2.readValue("{\"height\":24}", AnotherBean.class);
-        
+
         // Assert
         // 4. Should have created two cache instance
         assertEquals(2, cacheProvider.createCacheCount());
     }
 
     @Test
-    public void testBuilderBuildWithDefaults() throws Exception
+    public void testBuilderValueValidation() throws Exception
     {
         // does not throw
-        DefaultCacheProvider.builder().build();
+        DefaultCacheProvider.builder()
+                .build();
+        DefaultCacheProvider.builder()
+                .maxDeserializerCacheSize(Integer.MAX_VALUE)
+                .maxSerializerCacheSize(Integer.MAX_VALUE)
+                .build();
+
+        // validation fails
+        _verifyValidationFail(() -> DefaultCacheProvider.builder().maxDeserializerCacheSize(-1),
+                "Cannot set maxDeserializerCacheSize to a negative value");
+        _verifyValidationFail(() -> DefaultCacheProvider.builder().maxSerializerCacheSize(-1),
+                "Cannot set maxSerializerCacheSize to a negative value");
+    }
+
+    private void _verifyValidationFail(Runnable cacheProviderBuilder, String expectedErrorMsg) {
+        try {
+            cacheProviderBuilder.run();
+            fail("Should not reach here");
+        } catch (IllegalArgumentException e) {
+            assertTrue(e.getMessage().contains(expectedErrorMsg));
+        }
+    }
+
+    @Test
+    public void testDefaultCacheProviderConfigSerializerCache() throws Exception
```

Comment:
Not quite sure what is being tested here...

Reply:
Hmmmh. But it doesn't really check if it takes effect wrt changing cache size.
Just that nothing breaks in obvious way.

But I guess as sanity check, maybe extract common code that writes and reads output into helper method to reduce duplication.